### PR TITLE
update reqwest and check response status

### PR DESCRIPTION
**Pull request by jaemk**
- update reqwest crate
- make sure response status is checked
- only compile doc tests (don't run network things)
- `reqwest` feature doesn't exist (was blocking unit tests running)

### CODE REVIEW

**Comment by jaemk** [comment]
It seems like there's a bigger issue with parsing in general related to the `RelatedTopic` `enum` which expects the contents of `TopicResult` to be in a specific order. I'll open a separate issue for this.

**Comment by Aaronepower** [comment]
Thank you for the PR and congratulations on your first contribution! Once you rebase this on top of the new changes I'll be happy to merge it in.

File: src/query.rs [hunk 1]
```diff
@@ -35,13 +36,14 @@ impl<'a> Query<'a> {
...
     ///
-    /// ```
+    /// ```no_run
     /// use ddg::Query;
     /// const APP_NAME: &'static str = "ddg_example_app";
     ///
     /// let query = Query::new("Rust", APP_NAME).no_html();
     ///
     /// let response = query.execute().unwrap();
+    /// # let _r = response;
```
Comment (Aaronepower) [inline src/query.rs]: Why is this needed?

Comment (jaemk) [reply src/query.rs]: There was a unused variable warning before the doc examples were `no_run`. Forgot to remove it!

**Comment by jaemk** [comment]
rebased!

**Comment by Aaronepower** [comment]
Thanks again for the PR!
